docs/analysis/influxdb.md

```diff
@@ -0,0 +1,41 @@
+# InfluxDB Metrics
+
+An [InfluxDB](https://www.influxdata.com/) query using [Flux](https://docs.influxdata.com/influxdb/cloud/query-data/get-started/query-influxdb/) can be used to obtain measurements for analysis.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisTemplate
+metadata:
+  name: error-rate
+spec:
+  args:
+  - name: application-name
+  metrics:
+  - name: error-rate
+    # NOTE: To be consistent with the prometheus metrics provider InfluxDB query results are returned as an array.
+    # In the example we're looking at index 0 of the returned array to obtain the value we're using for the success condition
+    successCondition: result[0] <= 0.01
+    provider:
+      influxdb:
+        profile: my-influxdb-secret  # optional, defaults to 'influxdb'
```

Comment:
Do we imagine the need for multiple influxDB profiles within the same cluster? And would different users of the cluster specify different influxdb profiles?

My thought process is that the argo-rollouts admin would configure a single influxdb at a system-level, used by all rollouts in the cluster. In which case, the end-users would never need to specify a profile in their templates and we could eliminate this field in the analysis template completely.

Reply:
The use case I had in mind is that you may have multiple orgs that you want to query within a single Influx database, or multiple Influx cloud accounts that you may want to query from a single cluster - It's probably not necessary as each cluster rollouts is running in would likely have a single org/database to query - happy to simplify it there.

Reply:
Thanks for explaining the use case. I'm asking because we made a misstep with the prometheus provider by making the prometheus server address a required field in the template (which is not something dev teams are typically concerned with) as opposed to a central configuration. And so I wanted to avoid a similar mistake here. I actually just filed an issue to allow the prometheus address to be optional/centrally defined: https://github.com/argoproj/argo-rollouts/issues/1846.

When comparing the metric provider integrations there are a couple of different approaches of where connection information is configured:

Prometheus: analysis template level (too fined grain)
DataDog: single central definition
NewRelic: multiple central definitions (with a default)
Wavefront: multiple central definitions (with no default)
CloudWatch: single central definition

Your implementation is like NewRelic, which IMO the most flexible, so let's stick with your approach.

Reply:
That makes sense - thanks for the context and yeah I modelled after NewRelic. I also found it to be the most flexible - so I'll leave it as is.